src/indieweb_utils/replies/context.py

```diff
@@ -29,298 +48,306 @@ def get_reply_context(url, twitter_bearer_token=None):
 
-            return True, h_entry, site_supports_webmention
+        if post_video_url:
+            return_object.video = post_video_url
+        else:
+            return_object.video = ""
 
-        elif parsed["items"] and parsed["items"][0]["type"] == ["h-card"]:
-            h_card = parsed["items"][0]
+        return return_object
 
-            if h_card["properties"].get("name"):
-                author_name = h_card["properties"]["name"][0]
-            else:
-                author_name = None
-
-            if h_card["properties"].get("photo"):
-                author_image = h_card["properties"]["photo"][0]
-                if author_image.startswith("//"):
-                    author_image = "https:" + author_image
-                elif author_image.startswith("/"):
-                    author_image = url.split("/")[0] + "//" + domain + author_image
-                elif author_image.startswith("http://") or author_image.startswith("https://"):
-                    author_image = author_image
-                else:
-                    author_image = "https://" + domain + "/" + author_image
-            else:
-                author_image = None
+    elif parsed["items"] and parsed["items"][0]["type"] == ["h-card"]:
+        h_card = parsed["items"][0]
 
-            if h_card["properties"].get("note"):
-                post_body = h_card["properties"]["note"][0]
+        if h_card["properties"].get("name"):
+            author_name = h_card["properties"]["name"][0]
+        else:
+            author_name = ""
+
+        if h_card["properties"].get("photo"):
+            author_image = h_card["properties"]["photo"][0]
+            if author_image.startswith("//"):
+                author_image = "https:" + author_image
+            elif author_image.startswith("/"):
+                author_image = parsed_url.scheme + "://" + domain + author_image
+            elif author_image.startswith("http://") or author_image.startswith("https://"):
+                author_image = author_image
             else:
-                post_body = None
+                author_image = "https://" + domain + "/" + author_image
+        else:
+            author_image = ""
 
-            h_entry = {
-                "type": "entry",
-                "author": {"type": "card", "url": url, "name": author_name, "photo": author_image},
-                "url": url,
-                "content": {"html": post_body},
-            }
+        if h_card["properties"].get("note"):
+            post_body = h_card["properties"]["note"][0]
+        else:
+            post_body = ""
 
-            return True, h_entry, site_supports_webmention
+        return_object.name = ""
+        return_object.post_url = url
+        return_object.post_text = post_body
+        return_object.post_html = post_body
+        return_object.authors = [PostAuthor(url=url, name=author_name, photo=author_image)]
+        return_object.photo = ""
+        return_object.video = ""
```

Comment:
When using a dataclass, you should instantiate it instead of setting attributes on an un-instantiated class.

```python
return ReplyContext(
    name = ""
    post_url=post_url
    post_body=post_body
   ...
)
```